meson.build

```diff
@@ -214,7 +214,10 @@ endif
 libxmlb = dependency('xmlb', version : '>= 0.1.13', fallback : ['libxmlb', 'libxmlb_dep'])
 gusb = dependency('gusb', version : '>= 0.3.5', fallback : ['gusb', 'gusb_dep'])
 sqlite = dependency('sqlite3')
-libarchive = dependency('libarchive')
+if get_option('libarchive')
+  libarchive = dependency('libarchive')
+  conf.set('HAVE_LIBARCHIVE', '1')
+endif
```

Comment:
rather than a new option, how about instead make it like this:
```
libarchive = dependency('libarchive', required: false)
if libarchive.found()
    conf.set('HAVE_LIBARCHIVE', '1')
else
   warning('buliding without libarchive support')
endif
```

Reply:
I think the gentoo/arch crowd get very upset when you do this; https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Automagic_dependencies

Reply:
Then  I guess this should be correct PR then until #2738 goes in, in which case I think you really want to make it a requirement for anything but a fuzzing build.